### ISSUE TITLE
[MRESOURCES-43] resources:resources overwrites destination files when filtering is on, not when filtering is off

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-resources-plugin#168]
**[Roar Lauritzsen](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=roarl)** opened **[MRESOURCES-43](https://issues.apache.org/jira/browse/MRESOURCES-43?redirect=false)** and commented

It is not possible to specify that resources:resources should overwrite destination files or not if they already exist.
However, the behavior should at the very least be predictable so that one can achieve the desired result through phase ordering or ant scripts.
As it is today, when filtering is turned on, destination files are overwritten. When filtering is off, destination files are not overwritten.
If not selectable, the most practical solution would be to always overwrite (I believe).

---

**Affects:** 2.2

**Issue Links:**
- [MRESOURCES-21](https://issues.apache.org/jira/browse/MRESOURCES-21) when pom specifies multiple resource directories, the directories should be copied in the order they are in the pom
 (_**"duplicates"**_)
- [MRESOURCES-68](https://issues.apache.org/jira/browse/MRESOURCES-68) resources:resources overwrites filtered resources in target directory even if unchanged
 (_**"is duplicated by"**_)

2 votes, 2 watchers

**Comment by jira-importer** [comment]
**[Bart Robeyns](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bartrobeyns)** commented

I've created a patch that allows you to set a 'forceWrite'-flag on the resources-plugin's configuration; it defaults to 'false', maintaining the current behaviour.
The patch removes the destination-file before copying the source file over.
(I encountered the issue myself when weaving dependent jars with the aspectj-plugin; resources in the dependent jars get extracted in target\classes, and afterwards the resources-plugin won't override them anymore.)

However, the actual ambiguity lies in Plexus' FileUtils.copyFile, which only compares the modified-date of source and destinations if filtering is off;
it might be more appropriate to provide a flag in this method whether the date should be verified (both with and without filtering turned on).
Hence my question: should I upload the patch or not?
